File: hack_v2/types/company.ts
export interface Contact {
  id: string;
  name: string;
  role: string | null;
  email: string | null;
  linkedin: string | null;
  source?: string;
}

export interface Company {
  id: string;
  company: string;
  website: string;
  industry: string;
  employees: number | null;
  trigger: string;
  score: number;
  summary: string;
  contacts: Contact[];
  qualified?: boolean;
  trigger_source?: string | null;
  trigger_confidence?: number | null;
  contact_confidence?: number | null;
  summary_confidence?: number | null;
  status?: string | null;
  created_at?: string | null;
  firecrawl_used?: boolean;
  news_used?: boolean;
  news_headlines?: string | null;
  discovery_timestamp?: string | null;
  sales_playbook?: {
    lead_priority?: string;
    why_qualified?: string;
    buying_signals?: string;
    supporting_evidence?: string;
    recommended_decision_makers?: string;
    outreach_strategy?: string;
    communication_channel?: string;
    followup_timeline?: string;
    subject_line?: string;
    cold_email_opening?: string;
    next_best_action?: string;
  } | null;
  icp_breakdown?: {
    industry_match: number;
    company_size: number;
    buying_trigger: number;
    news_relevance: number;
    website_match: number;
    decision_makers_found: number;
  } | null;
  discovery_confidence?: number | null;
  evidence_sources?: string[] | null;
  pipeline_execution?: {
    [key: string]: {
      status: "completed" | "skipped" | "failed" | "running";
      duration: number;
    };
  } | null;
  execution_time?: number | null;
}

File: backend/models/company.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, Float, Boolean, DateTime, JSON
from sqlalchemy import UUID
from database import Base

from sqlalchemy.orm import relationship

class Company(Base):
    __tablename__ = "companies"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String, nullable=True)
    website = Column(String, unique=True, nullable=True)
    industry = Column(String, nullable=True)
    employee_count = Column(Integer, nullable=True)
    
    # Trigger Info
    trigger_type = Column(String, nullable=True)
    trigger_source = Column(String, nullable=True)
    trigger_confidence = Column(Float, nullable=True)
    contact_confidence = Column(Float, nullable=True)
    summary_confidence = Column(Float, nullable=True)
    
    # ICP Info
    icp_score = Column(Float, nullable=True)
    qualified = Column(Boolean, nullable=True)
    
    # Content
    summary = Column(String, nullable=True)
    
    # Pipeline State
    status = Column(String, nullable=True, default="discovered")
    
    # Firecrawl & News API integration fields
    firecrawl_used = Column(Boolean, nullable=True, default=False)
    news_used = Column(Boolean, nullable=True, default=False)
    news_headlines = Column(String, nullable=True)
    discovery_timestamp = Column(DateTime, default=datetime.utcnow)
    
    # Premium features fields
    sales_playbook = Column(JSON, nullable=True)
    icp_breakdown = Column(JSON, nullable=True)
    discovery_confidence = Column(Float, nullable=True)
    evidence_sources = Column(JSON, nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    contacts = relationship("Contact", back_populates="company", cascade="all, delete-orphan", lazy="joined")

